airbyte-cdk/python/airbyte_cdk/sources/streams/concurrent/state_converters/abstract_stream_state_converter.py

```diff
@@ -19,11 +19,60 @@ class AbstractStreamStateConverter(ABC):
     END_KEY = "end"
 
     @abstractmethod
+    def _from_state_message(self, value: Any) -> Any:
+        pass
+
+    @abstractmethod
+    def _to_state_message(self, value: Any) -> Any:
+        pass
+
+    def __init__(self, is_sequential_state: bool = True):
```

Comment:
This is needed as the fact that the state is sequential or not is part of this class and now that we want non-sequential state, this class needs to know about it